File: karma/test/calculator.spec.js
describe('Calculator', function() {
    let calculator;
    
    beforeEach(function() {
        calculator = new Calculator();
    });
    
    it('should add two numbers correctly', function() {
        expect(calculator.add(2, 3)).toBe(5);
    });
    
    it('should subtract two numbers correctly', function() {
        expect(calculator.subtract(5, 2)).toBe(3);
    });

    // Failing test
    it('should multiply two numbers correctly', function() {
        expect(calculator.multiply(2, 3)).toBe(10); // Intentional failure
    });

    // Flaky test
    it('should return a random number between 1 and 10', function() {
        const result = calculator.randomNumber();
        expect(result).toBeGreaterThanOrEqual(1);
        expect(result).toBeLessThanOrEqual(10);
        // Simulate flakiness
        if (Math.random() > 0.5) {
            expect(result).toBe(11); // Random failure
        }
    });
});

File: rspec/spec/calculator_spec.rb
require 'spec_helper'
require_relative '../lib/calculator'

RSpec.describe Calculator do
  let(:calculator) { Calculator.new }

  describe '#add' do
    it 'adds two positive numbers correctly' do
      expect(calculator.add(2, 3)).to eq(5)
    end

    it 'adds a positive and a negative number correctly' do
      expect(calculator.add(2, -3)).to eq(-1)
    end

    it 'adds two negative numbers correctly' do
      expect(calculator.add(-2, -3)).to eq(-5)
    end
  end

  describe '#subtract' do
    it 'subtracts two positive numbers correctly' do
      expect(calculator.subtract(5, 3)).to eq(2)
    end

    it 'subtracts a negative number from a positive number correctly' do
      expect(calculator.subtract(5, -3)).to eq(8)
    end

    it 'subtracts a positive number from a negative number correctly' do
      expect(calculator.subtract(-5, 3)).to eq(-8)
    end
  end
end

